packages/playground/remote/src/lib/boot-playground-remote.ts

```diff
@@ -232,6 +239,25 @@ export async function bootPlaygroundRemote() {
 		throw e;
 	}
 
+	/**
+	 * If the browser is online we can download WordPress assets asynchronously to speed up the boot process.
+	 * Missing assets will be fetched on demand from the Playground server until they are downloaded.
+	 *
+	 * If the browser is offline, we download WordPress assets synchronously to ensure the Playground is fully functional on load.
```

Comment:
How are we downloading assets if the browser is offline?

Also, nitpick: Synchronous means blocking, meaning the event loop would be stopped until the download is complete.

Reply:
The comment was misleading, this is what it needs to say: _ If the browser is offline, we backfill WordPress assets synchronously from the cache to ensure Playground is fully functional before boot finishes._

Reply:
I'm reopening this conversation becase this isn't addressed yet:

> Synchronous means blocking, meaning the event loop would be stopped until the download is complete. In here, however, we're not pausing the entire worker but just awaiting the download before continuing the boot process.

Reply:
I get it now. In my mind, the comment sounded the same, but await and synchronous definitely aren't the same. [3e4cd07](https://github.com/WordPress/wordpress-playground/pull/1604/commits/3e4cd07ccc82009406b85f8888a0c9dc7e1b5059)